packages/core/src/modules/proofs/services/ProofService.ts

```diff
@@ -908,7 +910,9 @@ export class ProofService {
     const credentialObjects = [
       ...Object.values(requestedCredentials.requestedAttributes),
       ...Object.values(requestedCredentials.requestedPredicates),
-    ].map((c) => c.credentialInfo)
+    ]
+      .filter(Boolean)
```

Comment:
It's a bit harder to understand what we want to achieve here with the `Boolean`. Does it check whether the value is not empty?

Reply:
Yes. It uses the boolean constructor to check for truthy/falsy

```js
[].filter((c) => c !== undefined)
```
Would be more verbose though.

Reply:
I see. You're right that it's more verbose. But it better expresses the intent.

I think the `null` could also be the case, right? Then it should be
```
[].filter((c) => c !== undefined && c !== null)
```

Then I wouldn't be afraid of creating new function `isNil` or `isNotNil` and extract this logic.